digital_asset_client.go

```diff
@@ -0,0 +1,553 @@
+	functionName := "mint"
+
+	// mint_soul_bound requires the recipient address as the last argument
+	args := [][]byte{
+		collectionBytes,
+		descriptionBytes,
+		nameBytes,
+		uriBytes,
+		propertyKeysBytes,
+		propertyTypesBytes,
+		propertyValuesBytes,
+	}
+
+	return functionName, args
+}
+
+// TransferToken transfers a token from one account to another
+func (dac *DigitalAssetClient) TransferToken(sender *Account, tokenAddress AccountAddress, recipient AccountAddress) (*string, error) {
+	// Serialize arguments
+	tokenAddressBytes, err := bcs.Serialize(&tokenAddress)
+	if err != nil {
+		return nil, fmt.Errorf("failed to serialize token address: %w", err)
+	}
+
+	recipientBytes, err := bcs.Serialize(&recipient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to serialize recipient: %w", err)
+	}
+
+	// Build the transaction
+	rawTxn, err := dac.client.BuildTransaction(sender.AccountAddress(), TransactionPayload{
+		Payload: &EntryFunction{
+			Module: ModuleId{
+				Address: AccountOne, // 0x1
+				Name:    "object",
+			},
+			Function: "transfer",
+			ArgTypes: []TypeTag{ObjectTypeTag},
+			Args: [][]byte{
+				tokenAddressBytes,
+				recipientBytes,
+			},
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to build transaction: %w", err)
+	}
+
+	// Sign and submit the transaction
+	signedTxn, err := rawTxn.SignedTransaction(sender)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign transaction: %w", err)
+	}
+
+	submitResult, err := dac.client.SubmitTransaction(signedTxn)
+	if err != nil {
+		return nil, fmt.Errorf("failed to submit transaction: %w", err)
+	}
+
+	return &submitResult.Hash, nil
+}
+
+// === HELPER FUNCTIONS FOR BCS SERIALIZATION ===
+
+// SerializeVectorString serializes a vector of strings for BCS
+
+func SerializeVectorString(strings []string) ([]byte, error) {
+	return bcs.SerializeSingle(func(ser *bcs.Serializer) {
+		bcs.SerializeSequenceWithFunction(strings, ser, func(ser *bcs.Serializer, item string) {
+			ser.WriteString(item)
+		})
+	})
+}
+
+// SerializeVectorBytes serializes a vector of byte arrays for BCS
+func SerializeVectorBytes(values []string) ([]byte, error) {
+	return bcs.SerializeSingle(func(ser *bcs.Serializer) {
+		byteArrays := make([][]byte, len(values))
+		for i, value := range values {
+			byteArrays[i] = []byte(value)
+		}
+
+		bcs.SerializeSequenceWithFunction(byteArrays, ser, func(ser *bcs.Serializer, item []byte) {
+			ser.WriteBytes(item)
+		})
+	})
+}
+
+func SerializePropertyValues(values []string, types []string) ([]byte, error) {
+	if len(values) != len(types) {
+		return nil, errors.New("property values and types must have the same length")
+	}
+
+	// Convert each value to BCS-serialized bytes according to its type
+	byteArrays := make([][]byte, len(values))
+
+	for i, value := range values {
+		propertyType := types[i]
+
+		var valueBytes []byte
+		var err error
+
+		switch propertyType {
```

Comment:
You should be able to use the `convertArg` to convert it into the correct type for each.  Then we don't need copied serialization methods